proxy/src/coco/source.rs

```diff
@@ -419,12 +421,26 @@ pub fn commit<'repo>(browser: &mut Browser<'repo>, sha1: &str) -> Result<Commit,
         }
     }
 
+    let branches = browser.revision_branches(sha1)?;
+
+    // If a commit figures in more than one branch, there's no real way to know
+    // which branch to show without additional context. So, we choose the first
+    // branch.
+    let branch = branches.first();
+
+    // Known commits always have at least one branch. If this isn't the case, it's a bug.
+    let branch = branch
+        .expect("known commits must be on a branch")
```

Comment:
I'd prefer we added an error variant than `expect`

Reply:
Is this an error we want to handle though? If a commit doesn't have a branch, I'm not sure what we're supposed to do?

Reply:
Well, our error variant gets resurfaced as a `warp::Rejection`. It's more to have a structure to the error and we're already in the `Result<_, error::Error>`